### PR TITLE
fix connection leakage

### PR DESCRIPTION
**Pull request by gangtao**
fix the nats connection leakage issue

### CODE REVIEW

**Comment by jovezhong** [comment]
Please check with @ye11ow / @zliang-min 

It looks that this change can be contributed back to benthos

**Comment by jovezhong** [comment]
we may not need to merge this. Since the upstream contains this fix https://github.com/benthosdev/benthos/pull/2347

**Comment by gangtao** [comment]
> we may not need to merge this. Since the upstream contains this fix [benthosdev#2347](https://github.com/benthosdev/benthos/pull/2347)

no, I believe we still need merge this one, as we are building on our own branch @ye11ow ?

File: internal/impl/nats/input.go [hunk 1]
```diff
@@ -251,6 +251,9 @@ func (n *natsReader) Read(ctx context.Context) (*service.Message, service.AckFun
 }
 
 func (n *natsReader) Close(ctx context.Context) (err error) {
+	go func() {
```
Comment (ye11ow) [inline internal/impl/nats/input.go:254]: is there any particular reason we need to wrap it inside a go routine? I think we'd better to just remove the go routine to make it a sync function.

**Review by ye11ow — APPROVED** [review]

**Comment by ye11ow** [comment]
Yes, we need to merge this one. Same issue with proton-go-driver, we have not sync with upstream for a while.

File: internal/impl/nats/input.go (hunk 1 above)
Comment (gangtao) [reply internal/impl/nats/input.go:254]: I read the jetstream code here https://github.com/timeplus-io/benthos/blob/main/internal/impl/nats/input_jetstream.go#L373 where the writer put the close logic inside the goroutine.

**Comment by jovezhong** [comment]
(Jove Github Bot) added it to the current sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) moved this ticket out of the GitHub project(up to 1200 tickets for one project).

**Pull request merged**